tests/integrationv2/providers.py

```diff
@@ -248,6 +275,55 @@ def _cipher_to_cmdline(self, protocol, cipher):
 
         return cmdline + ciphers
 
+    @classmethod
+    def supports_max_frag(cls):
+        if '1.0.2' in OpenSSL.get_version():
+            return False
+
+        return True
+
+    @classmethod
+    def get_version(cls):
+        return cls._version
+
+    @classmethod
+    def supports_protocol(cls, protocol, with_cert=None):
+        if protocol is Protocols.TLS13 and '1.0.2' in OpenSSL.get_version():
+            return False
+
+        return True
+
+    @classmethod
+    def supports_cipher(cls, cipher, with_curve=None):
+        is_openssl_111 = "openssl-1.1.1" in OpenSSL.get_version()
+        if is_openssl_111 and cipher.openssl1_1_1 is False:
+            return False
+
+        if not is_openssl_111:
+            # OpenSSL 1.0.2 does not have ChaChaPoly
+            if 'CHACHA20' in cipher.name:
+                return False
+
+        if cipher.fips is False and "fips" in OpenSSL.get_version():
+            return False
+
+        if "1.0.2" in OpenSSL.get_version() and with_curve is not None:
+            invalid_ciphers = [
+                Ciphers.ECDHE_RSA_AES128_SHA,
+                Ciphers.ECDHE_RSA_AES256_SHA,
+                Ciphers.ECDHE_RSA_AES128_SHA256,
+                Ciphers.ECDHE_RSA_AES256_SHA384,
+                Ciphers.ECDHE_RSA_AES128_GCM_SHA256,
+                Ciphers.ECDHE_RSA_AES256_GCM_SHA384,
+            ]
+
+            # OpenSSL 1.0.2 and 1.0.2-FIPS can't find a shared cipher with S2N
+            # when P-384 is used, but I can't find any reason why.
```

Comment:
Maybe open an issue to track this investigation? We should really probably know why we're excluding these cases.

Reply:
#2130

Reply:
#2130